src/main/java/com/flab/offcoupon/service/StatisticsService.java

```diff
@@ -0,0 +1,107 @@
+package com.flab.offcoupon.service;
+
+import com.flab.offcoupon.domain.vo.persistence.statistics.MonthlyOrderStatisticsVo;
+import com.flab.offcoupon.domain.vo.persistence.statistics.MonthlyStatisticsParameterVo;
+import com.flab.offcoupon.dto.request.StatisticsRequest;
+import com.flab.offcoupon.dto.response.MonthlyOrderStatistics;
+import com.flab.offcoupon.exception.statistics.LocalDateBadRequestException;
+import com.flab.offcoupon.repository.mysql.StatisticsRepository;
+import com.flab.offcoupon.util.ResponseDTO;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.time.YearMonth;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static com.flab.offcoupon.exception.statistics.StatisticsErrorMessage.START_MUST_BE_BEFORE_THANT_END;
+
+@RequiredArgsConstructor
+@Service
+public class StatisticsService {
+
+    private final StatisticsRepository statisticsRepository;
+
+    /**
+     * 인덱스를 사용해서 쿼리 최적화를 했지만, where절에 해당하는 레코드가 많을 경우 성능이 떨어질 수 있습니다.<br>
+     * 따라서 병렬 스트림을 사용하여 월별로 쿼리를 실행하고 결과를 합치는 방식을 사용했습니다.
+     *
+     * @param request 월별 주문 통계 조회 요청
+     * @return 월별 주문 통계 조회 결과
+     */
+
+    @Transactional(readOnly = true)
+    public ResponseDTO<List<MonthlyOrderStatistics>> getMonthlyOrderStatistics(StatisticsRequest request) {
+        LocalDate startedAt = request.getStartedAt();
+        LocalDate endedAt = request.getEndedAt();
+        validateStartDateIsBeforeEndDate(startedAt, endedAt);
+        List<MonthlyOrderStatistics> monthlyStatisticsList = new ArrayList<>();
+        getMonthlyStatistics(startedAt, endedAt, monthlyStatisticsList);
```

Comment:
파라미터로 넘어간 `monthlyStatisticsList`에 값을 추가하지 말고, 통계결과를 List로 반환하게 만들어보세요. 파라미터의 상태를 바꾸는 함수나 메서드는 결과를 예측하기 어려울 뿐 아니라 상태를 바꾸는 사이드이펙트가 있는 메서드가 됩니다.

```suggestion
        List<MonthlyOrderStatistics> monthlyStatisticsList = getMonthlyStatistics(startedAt, endedAt);
```

Reply:
넵 반환하는걸로 변경하겠습니다 알려주셔서 감사합니다!